lib/hammer_cli_foreman_discovery/discovery.rb

```diff
@@ -99,7 +97,7 @@ def self.build_options
         option "--provision-method", "METHOD", " ",
                     :format => HammerCLI::Options::Normalizers::Enum.new(['build', 'image'])
 
-        super :without => [:root_pass, :ptable_id, :puppet_class_ids, :host_parameters_attributes]
```

Comment:
This should be present here until it's removed from API docs: https://github.com/theforeman/foreman_discovery/blob/012851e7cf1423db1f65c0e350afac989141d9ab/app/controllers/api/v2/discovered_hosts_controller.rb#L60

Also, I guess we should add here those (this will ensure that even if those params are documented in the foreman_discovery controller, they won't be built by this plugin. This means, that those 4 options should be added from h-c-f-puppet extension):
 - https://github.com/theforeman/foreman_discovery/blob/012851e7cf1423db1f65c0e350afac989141d9ab/app/controllers/api/v2/discovered_hosts_controller.rb#L54
 - https://github.com/theforeman/foreman_discovery/blob/012851e7cf1423db1f65c0e350afac989141d9ab/app/controllers/api/v2/discovered_hosts_controller.rb#L59
  - https://github.com/theforeman/foreman_discovery/blob/012851e7cf1423db1f65c0e350afac989141d9ab/app/controllers/api/v2/discovered_hosts_controller.rb#L70
  
But before doing so, @lzap, can you confirm that those params won't do anything unless foreman_puppet is installed?

Reply:
Just a quick update - discovery API is currently broken completely, need to fix it first and then decide, probably I will do "if defined puppet then include these attributes".